parse (and ignore for now) tensor cell type in tensor type spec

havardpe opened the pull request:
example: `tensor<float>(x[10])`

@arnej27959 please review

havardpe committed:
parse (and ignore for now) tensor cell type in tensor type spec

example: tensor<float>(x[10])

baldersheim reviewed (changes requested):
eval_node_types_test_app not too happy.

havardpe committed:
Improve handling of malformed cell type specifiers

Do not consume any input when encountering a malformed cell type
specifier. This avoids parse conflicts between 'less' operators and
tensor cell type specifiers in type expressions in the type resolution
test which parses partial input into value types.

example expression: 'tensor < double'

havardpe commented:
test should be more happy now

arnej27959 reviewed (approved):
looks good to me
